bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/BridgeRemoteMyElectricalDataHandler.java

```diff
@@ -0,0 +1,220 @@
+            + "v1/oauth2/authorize?client_id=%s&response_type=code&redirect_uri=na&user_type=na&state=na&person_id=-1&usage_points_id=%s";
+
+    public static final String LINKY_MYELECTRICALDATA_CLIENT_ID = "_h7zLaRr2INxqBI8jhDUQXsa_G4a";
+
+    private static final String TEMPO_URL = BASE_URL + "rte/tempo/%s/%s";
+
+    private static final DateTimeFormatter API_DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private static final DateTimeFormatter API_DATE_FORMAT_YEAR_FIRST = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    // https://www.myelectricaldata.fr/v1/oauth2/authorize?response_type=code&client_id=&state=linky&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Fconnectlinky&scope=am_application_scope+default&user_type=aa&person_id=-1&usage_points_id=aa
+
+    public BridgeRemoteMyElectricalDataHandler(Bridge bridge, final @Reference HttpClientFactory httpClientFactory,
+            final @Reference OAuthFactory oAuthFactory, final @Reference HttpService httpService,
+            final @Reference ThingRegistry thingRegistry, ComponentContext componentContext, Gson gson) {
+        super(bridge, httpClientFactory, oAuthFactory, httpService, thingRegistry, componentContext, gson);
+    }
+
+    @Override
+    public void connectionInit() {
+        connected = true;
+    }
+
+    @Override
+    public void initialize() {
+        tokenUrl = BridgeRemoteMyElectricalDataHandler.LINKY_MYELECTRICALDATA_API_TOKEN_URL;
+        authorizeUrl = BridgeRemoteMyElectricalDataHandler.LINKY_MYELECTRICALDATA_AUTHORIZE_URL;
+
+        super.initialize();
+    }
+
+    @Override
+    public String getClientId() {
+        return BridgeRemoteMyElectricalDataHandler.LINKY_MYELECTRICALDATA_CLIENT_ID;
+    }
+
+    @Override
+    public String getClientSecret() {
+        return "";
+    }
+
+    @Override
+    public boolean getIsSandbox() {
+        return false;
+    }
+
+    @Override
+    public String formatAuthorizationUrl(String redirectUri) {
+        return super.formatAuthorizationUrl("");
+    }
+
+    @Override
+    public String authorize(String redirectUri, String reqState, String reqCode) throws LinkyException {
+        String url = String.format(BridgeRemoteMyElectricalDataHandler.LINKY_MYELECTRICALDATA_API_TOKEN_URL,
+                getClientId(), reqCode);
+        EnedisHttpApi enedisApi = getEnedisApi();
+        if (enedisApi == null) {
+            return "";
+        }
+        String token = enedisApi.getContent(url);
+
+        logger.debug("token: {}", token);
+
+        Collection<Thing> col = this.thingRegistry.getAll();
+
+        for (Thing thing : col) {
+            if (LinkyBindingConstants.THING_TYPE_LINKY.equals(thing.getThingTypeUID())) {
+                Configuration config = thing.getConfiguration();
+                String prmId = (String) config.get("prmId");
+
+                if (!prmId.equals(reqCode)) {
+                    continue;
+                }
```

Comment:
That means prmId is kind of mandatory for this bridge.
If not provided, the token in the thing configuration will never be updated.